AutomationHelpers/src/UserCodeCollections/WebLibrary.cs

```diff
@@ -0,0 +1,63 @@
+﻿/*
+ * Created by Ranorex
+ * User: sknopper
+ * Date: 9/21/2017
+ * Time: 12:17 PM
+ * 
+ * To change this template use Tools | Options | Coding | Edit Standard Headers.
+ */
+using System;
+using System.Collections.Generic;
+using System.Drawing;
+using System.IO;
+using System.Net;
+using System.Text;
+using System.Text.RegularExpressions;
+using System.Threading;
+
+using WinForms = System.Windows.Forms;
+using Ranorex;
+using Ranorex.Core;
+using Ranorex.Core.Testing;
+
+namespace Ranorex.AutomationHelpers.UserCodeCollections
+{
+    /// <summary>
+    /// Collection of methods which helps to automate web and network based tasks.
+    /// </summary>
+    [UserCodeCollection]
+    public class WebLibrary
+    {
+    	
+    	/// <summary>
+        /// Download a file and stores it locally.
+        /// </summary>
+        /// <param name="uri">The uri of the file to download</param>
+        /// <param name="targetLocation">Local location where the file should be stored</param>
+        /// <param name="overwrite">true / false if existing file should be overwritten</param>
+    	[UserCodeMethod]
+    	public static void DownloadFile(string uri, string localPath, string overwriteExisting = "true")
+    	{
+    		if (!File.Exists(localPath) || bool.Parse(overwriteExisting)){
+    			using (WebClient client = new WebClient()) 
+    			{
+    				try
+    				{
+    					client.DownloadFile(uri, localPath);
+    					string message = String.Format("File successfully downloaded to {0}", new FileInfo(localPath).FullName);
+    					Report.Log(ReportLevel.Success, message);
+    				} catch (Exception e)
+    				{
+    					string message = String.Format("Downloading a file from: {0} failed faile for the following reason: {1}", uri, e.Message);
+    					Report.Log(ReportLevel.Failure, message);
+    				}
+    				
+    			}
+    		} else {
+                //TODO: what should happen if file exists and we don't want to overwrite it?
```

Comment:
You also just log that the download failed, so this is consistent.
But IMHO throwing and exception or passing on the exception would be more correct, since following code will most probably rely on the downloaded file.
Additionally, the method does not provide a return value, so you don't even know in code that it succeeded or failed, so you cannot react to that -> IMHO go with exception

Reply:
Its not that easy to go with an exception. The execution should only get into this branch if the file exists (was already downloaded before) and the test creator has decided to not download this file again. In this case the test should also be positive because the file exists and is ready for use.

Please feel free to correct me if i am wrong.